scripts/inference.py

```diff
@@ -173,4 +173,5 @@ def infer(use_cache, do_sample):
     False
 ]  # True/False are identical with greedy iff `torch.use_deterministic_algorithms(True)`
 for sample, cache in itertools.product(do_sample, use_cache):
-    infer(cache, sample)
+    with torch.no_grad():
```

Comment:
this is fine but we could also consider applying universally further up in this script with:

```
model.eval()
torch.set_grad_enabled(False)
```
as we did in the benchmark_inference.py script.